crates/cheatcodes/src/test.rs

```diff
@@ -34,20 +34,29 @@ impl Cheatcode for zkRegisterContractCall {
             zkDeployedBytecode,
         } = self;
 
-        let new_contract = DualCompiledContract {
-            name: name.clone(),
-            zk_bytecode_hash: zkBytecodeHash.0.into(),
-            zk_deployed_bytecode: zkDeployedBytecode.clone(),
+        let new_contract = DualCompiledContract::new(
+            name.clone(),
+            None,
+            zkBytecodeHash.0.into(),
+            zkDeployedBytecode.clone(),
             //TODO: add argument to cheatcode
-            zk_factory_deps: vec![],
-            evm_bytecode_hash: *evmBytecodeHash,
-            evm_deployed_bytecode: evmDeployedBytecode.clone(),
-            evm_bytecode: evmBytecode.clone(),
-        };
+            vec![],
+            *evmBytecodeHash,
+            evmDeployedBytecode.clone(),
+            evmBytecode.clone(),
+        );
 
         if let Some(existing) = ccx.state.dual_compiled_contracts.iter().find(|contract| {
-            contract.evm_bytecode_hash == new_contract.evm_bytecode_hash &&
-                contract.zk_bytecode_hash == new_contract.zk_bytecode_hash
+            contract
+                .evm
+                .as_ref()
+                .map(|evm| evm.bytecode_hash == *evmBytecodeHash)
+                .unwrap_or_default() &&
+                contract
+                    .zk
+                    .as_ref()
+                    .map(|zk| zk.bytecode_hash == zkBytecodeHash.0.into())
+                    .unwrap_or_default()
```

Comment:
I do agree that the interface is much harder. I think this is the path of least resistance to remove the notion that we need BOTH solc and zksolc. 
Having both as option came from the start of the refactor, when I was trying to also collect zksolc contracts that (for some reason) didn't have a solc equivalent, so when we did a lookup we could at least identify what contract it was instead of spewing some bytecode.
We can make this a bit easier by forcing EVM to always be present (removing the option), but we keep the optional on zksolc, allowing us to still create a `DualCompiledContract` that's valid when we don't compile with zksolc

Reply:
Removed these changes - we can improve the error message on lookup failure in another PR